tests/functional/adapter/incremental/fixtures.py

```diff
@@ -279,6 +309,140 @@
 {% endif %}
 """
 
+skip_matched_model = """
+{{ config(
+    materialized = 'incremental',
+    unique_key = 'id',
+    incremental_strategy='merge',
+    skip_matched_step = true,
+) }}
+
+{% if not is_incremental() %}
+
+-- data for first invocation of model
+
+select 1 as id, 'hello' as msg, 'blue' as color
+union all
+select 2 as id, 'goodbye' as msg, 'red' as color
+
+{% else %}
+
+-- data for subsequent incremental update
+
+select 1 as id, 'hey' as msg, 'cyan' as color
+union all
+select 2 as id, 'yo' as msg, 'green' as color
+union all
+select 3 as id, 'anyway' as msg, 'purple' as color
+
+{% endif %}
+"""
+
+skip_not_matched_model = skip_matched_model.replace(
+    "skip_matched_step = true", "skip_not_matched_step = true"
+)
+
+matching_condition_model = """
+{{ config(
+    materialized = 'incremental',
+    unique_key = 'id',
+    incremental_strategy='merge',
+    target_alias='t',
```

Comment:
Only `target_alias` is covered by the test, while actually we should test also `source_alias`